app/scripts/components/common/page-header/nav/nav-dropdown-button.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { USWDSNavDropDownButton } from '../../uswds/header/nav-drop-down-button';
+import { USWDSMenu } from '../../uswds/header/menu';
+import { DropdownNavLink } from '../types';
+import { createDynamicNavMenuList } from './create-dynamic-nav-menu-list';
+import { SetState } from '$types/aliases';
+import { LinkProperties } from '$types/veda';
+
+interface NavDropDownButtonProps {
+  item: DropdownNavLink;
+  isOpen: boolean[];
+  setIsOpen: SetState<boolean[]>;
+  index: number;
+  linkProperties: LinkProperties;
+}
+
+export const NavDropDownButton = ({
+  item,
+  isOpen,
+  setIsOpen,
+  index,
+  linkProperties
+}: NavDropDownButtonProps) => {
+  const onToggle = (index: number, setIsOpen: SetState<boolean[]>): void => {
+    setIsOpen((prevIsOpen) => {
+      const newIsOpen = prevIsOpen.map(
+        (prev, i) =>
+          i === index
+            ? !prev // toggle the current dropdown
+            : false // close all other dropdowns
+      );
```

Comment:
Noticed that `IsOpen` when true is persisting across navigations to other pages. We should probably have the dropdown close when a users mouse navigates outside of the dropdown or when a user clicks on another item in the navigation or page. I will leave that call up to @faustoperez. But that can be a follow up, ticket to not drag things out.  

At minimum we should set a default `false` value on render so that the menus don't persist open when users navigate to other pages. 

https://github.com/user-attachments/assets/65fb1327-ef7c-4db6-a124-226edb14a541

Reply:
this is already noted 👍🏼

Reply:
Opened a follow-up: https://github.com/NASA-IMPACT/veda-ui/issues/1296